File: menu-scanner-frontend-owner/src/features/platform-bank/store/models/response/platform-bank-response.ts
import { Pagination } from "@/utils/common/pagination";

export interface PlatformBankResponseModel {
  id: string;
  name: string;
  accountName: string;
  accountNumber: string;
  description?: string;
  status: string;
  displayOrder: number;
  image?: {
    sm?: string;
    md?: string;
    o?: string;
  } | null;
  createdAt?: string;
  updatedAt?: string;
}

export interface AllPlatformBankResponseModel extends Pagination {
  content: PlatformBankResponseModel[];
}

File: menu-scanner-backend/src/main/java/com/emenu/features/master/dto/response/PlatformBankResponse.java
package com.emenu.features.master.dto.response;

import com.emenu.enums.common.Status;
import com.emenu.shared.dto.ImageUrls;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlatformBankResponse {

    private UUID id;
    private String name;
    private String accountName;
    private String accountNumber;
    private String description;
    private Status status;
    private Integer displayOrder;
    private ImageUrls image;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
